rulewerk-core/src/main/java/org/semanticweb/rulewerk/core/reasoner/implementation/FileDataSource.java

```diff
@@ -54,32 +55,33 @@
 	/**
 	 * Constructor.
 	 *
-	 * @param file               a file that will serve as storage for fact terms.
+	 * @param filePath           path to a file that will serve as storage for fact
+	 *                           terms.
 	 * @param possibleExtensions a list of extensions that the files could have
 	 * @throws IOException              if the path of the given {@code file} is
 	 *                                  invalid.
 	 * @throws IllegalArgumentException if the extension of the given {@code file}
 	 *                                  does not occur in
 	 *                                  {@code possibleExtensions}.
 	 */
-	public FileDataSource(@NonNull final File file, final Iterable<String> possibleExtensions) throws IOException {
-		Validate.notNull(file, "Data source file cannot be null!");
-		final String fileName = file.getName();
-
-		this.file = file;
+	public FileDataSource(final String filePath, final Iterable<String> possibleExtensions) throws IOException {
+		Validate.notBlank(filePath, "Data source file name cannot be null!");
+
+		this.file = new File(filePath);
```

Comment:
Do we need to keep the File field after the constructor was called?

Reply:
It's still used in `equals()` and `hashCode()`, for Example.

Reply:
I see, I guess we still want to consider two FileDataSource objects with different path names that refer to the same file as equal.